{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\tqdm\\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n",
      "No path specified. Models will be saved in: \"AutogluonModels\\ag-20230821_225132\\\"\n",
      "Presets specified: ['best_quality']\n",
      "Stack configuration (auto_stack=True): num_stack_levels=3, num_bag_folds=10, num_bag_sets=20\n",
      "Warning: Training may take a very long time because `time_limit` was not specified and `train_data` is large (204000 samples, 35.61 MB).\n",
      "\tConsider setting `time_limit` to ensure training finishes within an expected duration or experiment with a small portion of `train_data` to identify an ideal `presets` and `hyperparameters` configuration.\n",
      "Beginning AutoGluon training ...\n",
      "AutoGluon will save models to \"AutogluonModels\\ag-20230821_225132\\\"\n",
      "AutoGluon Version:  0.8.2\n",
      "Python Version:     3.9.13\n",
      "Operating System:   Windows\n",
      "Platform Machine:   AMD64\n",
      "Platform Version:   10.0.22000\n",
      "Disk Space Avail:   25.20 GB / 511.33 GB (4.9%)\n",
      "Train Data Rows:    204000\n",
      "Train Data Columns: 13\n",
      "Label Column: 전력소비량(kWh)\n",
      "Preprocessing data ...\n",
      "Using Feature Generators to preprocess the data ...\n",
      "Fitting AutoMLPipelineFeatureGenerator...\n",
      "\tAvailable Memory:                    7751.52 MB\n",
      "\tTrain Data (Original)  Memory Usage: 33.98 MB (0.4% of available memory)\n",
      "\tInferring data type of each feature based on column values. Set feature_metadata_in to manually specify special dtypes of the features.\n",
      "\tStage 1 Generators:\n",
      "\t\tFitting AsTypeFeatureGenerator...\n",
      "\tStage 2 Generators:\n",
      "\t\tFitting FillNaFeatureGenerator...\n",
      "\tStage 3 Generators:\n",
      "\t\tFitting IdentityFeatureGenerator...\n",
      "\t\tFitting CategoryFeatureGenerator...\n",
      "\t\t\tFitting CategoryMemoryMinimizeFeatureGenerator...\n",
      "\t\tFitting DatetimeFeatureGenerator...\n",
      "\tStage 4 Generators:\n",
      "\t\tFitting DropUniqueFeatureGenerator...\n",
      "\tStage 5 Generators:\n",
      "\t\tFitting DropDuplicatesFeatureGenerator...\n",
      "\tUseless Original Features (Count: 1): ['year']\n",
      "\t\tThese features carry no predictive signal and should be manually investigated.\n",
      "\t\tThis is typically a feature which has the same value for all rows.\n",
      "\t\tThese features do not need to be present at inference time.\n",
      "\tUnused Original Features (Count: 1): ['num_date_time']\n",
      "\t\tThese features were not used to generate any of the output features. Add a feature generator compatible with these features to utilize them.\n",
      "\t\tFeatures can also be unused if they carry very little information, such as being categorical but having almost entirely unique values or being duplicates of other features.\n",
      "\t\tThese features do not need to be present at inference time.\n",
      "\t\t('object', []) : 1 | ['num_date_time']\n",
      "\tTypes of features in original data (raw dtype, special dtypes):\n",
      "\t\t('datetime', []) : 1 | ['일시']\n",
      "\t\t('float', [])    : 6 | ['기온(C)', '습도(%)', 'holidays', 'sin_time', 'cos_time', ...]\n",
      "\t\t('int', [])      : 4 | ['건물번호', 'hour', 'day', 'month']\n",
      "\tTypes of features in processed data (raw dtype, special dtypes):\n",
      "\t\t('float', [])                : 6 | ['기온(C)', '습도(%)', 'holidays', 'sin_time', 'cos_time', ...]\n",
      "\t\t('int', [])                  : 4 | ['건물번호', 'hour', 'day', 'month']\n",
      "\t\t('int', ['datetime_as_int']) : 2 | ['일시', '일시.dayofweek']\n",
      "\t1.4s = Fit runtime\n",
      "\t11 features in original data used to generate 12 features in processed data.\n",
      "\tTrain Data (Processed) Memory Usage: 19.58 MB (0.3% of available memory)\n",
      "Data preprocessing and feature engineering runtime = 1.51s ...\n",
      "AutoGluon will gauge predictive performance using evaluation metric: 'root_mean_squared_error'\n",
      "\tThis metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.\n",
      "\tTo change this, specify the eval_metric parameter of Predictor()\n",
      "User-specified model hyperparameters to be fit:\n",
      "{\n",
      "\t'NN_TORCH': {},\n",
      "\t'GBM': [{'extra_trees': True, 'ag_args': {'name_suffix': 'XT'}}, {}, 'GBMLarge'],\n",
      "\t'CAT': {},\n",
      "\t'XGB': {},\n",
      "\t'FASTAI': {},\n",
      "\t'RF': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'squared_error', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression', 'quantile']}}],\n",
      "\t'XT': [{'criterion': 'gini', 'ag_args': {'name_suffix': 'Gini', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'entropy', 'ag_args': {'name_suffix': 'Entr', 'problem_types': ['binary', 'multiclass']}}, {'criterion': 'squared_error', 'ag_args': {'name_suffix': 'MSE', 'problem_types': ['regression', 'quantile']}}],\n",
      "\t'KNN': [{'weights': 'uniform', 'ag_args': {'name_suffix': 'Unif'}}, {'weights': 'distance', 'ag_args': {'name_suffix': 'Dist'}}],\n",
      "}\n",
      "AutoGluon will fit 4 stack levels (L1 to L4) ...\n",
      "Fitting 11 L1 models ...\n",
      "Fitting model: KNeighborsUnif_BAG_L1 ...\n",
      "\t-2420.5984\t = Validation score   (-root_mean_squared_error)\n",
      "\t0.47s\t = Training   runtime\n",
      "\t0.68s\t = Validation runtime\n",
      "Fitting model: KNeighborsDist_BAG_L1 ...\n",
      "\t-2322.2389\t = Validation score   (-root_mean_squared_error)\n",
      "\t0.51s\t = Training   runtime\n",
      "\t0.68s\t = Validation runtime\n",
      "Fitting model: LightGBMXT_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMXT_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBM_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBM_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: RandomForestMSE_BAG_L1 ...\n",
      "\t-132.5674\t = Validation score   (-root_mean_squared_error)\n",
      "\t32.09s\t = Training   runtime\n",
      "\t4.11s\t = Validation runtime\n",
      "Fitting model: CatBoost_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused CatBoost_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: ExtraTreesMSE_BAG_L1 ...\n",
      "\t-167.4034\t = Validation score   (-root_mean_squared_error)\n",
      "\t10.88s\t = Training   runtime\n",
      "\t4.07s\t = Validation runtime\n",
      "Fitting model: NeuralNetFastAI_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetFastAI_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: XGBoost_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused XGBoost_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: NeuralNetTorch_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetTorch_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBMLarge_BAG_L1 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMLarge_BAG_L1 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: WeightedEnsemble_L2 ...\n",
      "\t-132.038\t = Validation score   (-root_mean_squared_error)\n",
      "\t1.76s\t = Training   runtime\n",
      "\t0.01s\t = Validation runtime\n",
      "Fitting 9 L2 models ...\n",
      "Fitting model: LightGBMXT_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMXT_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBM_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBM_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: RandomForestMSE_BAG_L2 ...\n",
      "\t-130.8924\t = Validation score   (-root_mean_squared_error)\n",
      "\t99.54s\t = Training   runtime\n",
      "\t5.78s\t = Validation runtime\n",
      "Fitting model: CatBoost_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused CatBoost_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: ExtraTreesMSE_BAG_L2 ...\n",
      "\t-128.4377\t = Validation score   (-root_mean_squared_error)\n",
      "\t16.64s\t = Training   runtime\n",
      "\t5.1s\t = Validation runtime\n",
      "Fitting model: NeuralNetFastAI_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetFastAI_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: XGBoost_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused XGBoost_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: NeuralNetTorch_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetTorch_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBMLarge_BAG_L2 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMLarge_BAG_L2 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: WeightedEnsemble_L3 ...\n",
      "\t-128.3946\t = Validation score   (-root_mean_squared_error)\n",
      "\t1.07s\t = Training   runtime\n",
      "\t0.0s\t = Validation runtime\n",
      "Fitting 9 L3 models ...\n",
      "Fitting model: LightGBMXT_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMXT_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBM_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBM_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: RandomForestMSE_BAG_L3 ...\n",
      "\t-132.5173\t = Validation score   (-root_mean_squared_error)\n",
      "\t68.11s\t = Training   runtime\n",
      "\t5.74s\t = Validation runtime\n",
      "Fitting model: CatBoost_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused CatBoost_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: ExtraTreesMSE_BAG_L3 ...\n",
      "\t-130.8139\t = Validation score   (-root_mean_squared_error)\n",
      "\t14.29s\t = Training   runtime\n",
      "\t4.92s\t = Validation runtime\n",
      "Fitting model: NeuralNetFastAI_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetFastAI_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: XGBoost_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused XGBoost_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: NeuralNetTorch_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetTorch_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBMLarge_BAG_L3 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMLarge_BAG_L3 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: WeightedEnsemble_L4 ...\n",
      "\t-130.7262\t = Validation score   (-root_mean_squared_error)\n",
      "\t1.12s\t = Training   runtime\n",
      "\t0.0s\t = Validation runtime\n",
      "Fitting 9 L4 models ...\n",
      "Fitting model: LightGBMXT_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMXT_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBM_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBM_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: RandomForestMSE_BAG_L4 ...\n",
      "\t-135.5223\t = Validation score   (-root_mean_squared_error)\n",
      "\t71.29s\t = Training   runtime\n",
      "\t6.11s\t = Validation runtime\n",
      "Fitting model: CatBoost_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused CatBoost_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: ExtraTreesMSE_BAG_L4 ...\n",
      "\t-133.358\t = Validation score   (-root_mean_squared_error)\n",
      "\t16.12s\t = Training   runtime\n",
      "\t5.2s\t = Validation runtime\n",
      "Fitting model: NeuralNetFastAI_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetFastAI_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: XGBoost_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused XGBoost_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: NeuralNetTorch_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused NeuralNetTorch_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: LightGBMLarge_BAG_L4 ...\n",
      "\tFitting 200 child models (S1F1 - S20F10) | Fitting with ParallelLocalFoldFittingStrategy\n",
      "\tWarning: Exception caused LightGBMLarge_BAG_L4 to fail during training... Skipping this model.\n",
      "\t\t[Errno 203] AssignProcessToJobObject() failed\n",
      "Detailed Traceback:\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1733, in _train_and_save\n",
      "    model = self._train_single(X, y, model, X_val, y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\trainer\\abstract_trainer.py\", line 1684, in _train_single\n",
      "    model = model.fit(X=X, y=y, X_val=X_val, y_val=y_val, total_resources=total_resources, **model_fit_kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\abstract\\abstract_model.py\", line 829, in fit\n",
      "    out = self._fit(**kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\stacker_ensemble_model.py\", line 169, in _fit\n",
      "    return super()._fit(X=X, y=y, time_limit=time_limit, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 266, in _fit\n",
      "    self._fit_folds(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\bagged_ensemble_model.py\", line 592, in _fit_folds\n",
      "    fold_fitting_strategy.after_all_folds_scheduled()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\autogluon\\core\\models\\ensemble\\fold_fitting_strategy.py\", line 508, in after_all_folds_scheduled\n",
      "    self.ray.init(**ray_init_args)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\client_mode_hook.py\", line 105, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\worker.py\", line 1451, in init\n",
      "    _global_node = ray._private.node.Node(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 287, in __init__\n",
      "    self.start_head_processes()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1124, in start_head_processes\n",
      "    self.start_monitor()\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\node.py\", line 1027, in start_monitor\n",
      "    process_info = ray._private.services.start_monitor(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 1911, in start_monitor\n",
      "    process_info = start_ray_process(\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\services.py\", line 877, in start_ray_process\n",
      "    ray._private.utils.set_kill_child_on_death_win32(process)\n",
      "  File \"C:\\Users\\wkdal\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python39\\site-packages\\ray\\_private\\utils.py\", line 963, in set_kill_child_on_death_win32\n",
      "    raise OSError(ctypes.get_last_error(), \"AssignProcessToJobObject() failed\")\n",
      "OSError: [Errno 203] AssignProcessToJobObject() failed\n",
      "Fitting model: WeightedEnsemble_L5 ...\n",
      "\t-133.3173\t = Validation score   (-root_mean_squared_error)\n",
      "\t0.92s\t = Training   runtime\n",
      "\t0.0s\t = Validation runtime\n",
      "AutoGluon training complete, total runtime = 438.34s ... Best model: \"WeightedEnsemble_L3\"\n",
      "Automatically performing refit_full as a post-fit operation (due to `.fit(..., refit_full=True)`\n",
      "Refitting models via `predictor.refit_full` using all of the data (combined train and validation)...\n",
      "\tModels trained in this way will have the suffix \"_FULL\" and have NaN validation score.\n",
      "\tThis process is not bound by time_limit, but should take less time than the original `predictor.fit` call.\n",
      "\tTo learn more, refer to the `.refit_full` method docstring which explains how \"_FULL\" models differ from normal models.\n",
      "Fitting model: KNeighborsUnif_BAG_L1_FULL | Skipping fit via cloning parent ...\n",
      "\t0.47s\t = Training   runtime\n",
      "\t0.68s\t = Validation runtime\n",
      "Fitting model: KNeighborsDist_BAG_L1_FULL | Skipping fit via cloning parent ...\n",
      "\t0.51s\t = Training   runtime\n",
      "\t0.68s\t = Validation runtime\n",
      "Fitting model: RandomForestMSE_BAG_L1_FULL | Skipping fit via cloning parent ...\n",
      "\t32.09s\t = Training   runtime\n",
      "\t4.11s\t = Validation runtime\n",
      "Fitting model: ExtraTreesMSE_BAG_L1_FULL | Skipping fit via cloning parent ...\n",
      "\t10.88s\t = Training   runtime\n",
      "\t4.07s\t = Validation runtime\n",
      "Fitting model: WeightedEnsemble_L2_FULL | Skipping fit via cloning parent ...\n",
      "\t1.76s\t = Training   runtime\n",
      "Fitting model: RandomForestMSE_BAG_L2_FULL | Skipping fit via cloning parent ...\n",
      "\t99.54s\t = Training   runtime\n",
      "\t5.78s\t = Validation runtime\n",
      "Fitting model: ExtraTreesMSE_BAG_L2_FULL | Skipping fit via cloning parent ...\n",
      "\t16.64s\t = Training   runtime\n",
      "\t5.1s\t = Validation runtime\n",
      "Fitting model: WeightedEnsemble_L3_FULL | Skipping fit via cloning parent ...\n",
      "\t1.07s\t = Training   runtime\n",
      "Fitting model: RandomForestMSE_BAG_L3_FULL | Skipping fit via cloning parent ...\n",
      "\t68.11s\t = Training   runtime\n",
      "\t5.74s\t = Validation runtime\n",
      "Fitting model: ExtraTreesMSE_BAG_L3_FULL | Skipping fit via cloning parent ...\n",
      "\t14.29s\t = Training   runtime\n",
      "\t4.92s\t = Validation runtime\n",
      "Fitting model: WeightedEnsemble_L4_FULL | Skipping fit via cloning parent ...\n",
      "\t1.12s\t = Training   runtime\n",
      "Fitting model: RandomForestMSE_BAG_L4_FULL | Skipping fit via cloning parent ...\n",
      "\t71.29s\t = Training   runtime\n",
      "\t6.11s\t = Validation runtime\n",
      "Fitting model: ExtraTreesMSE_BAG_L4_FULL | Skipping fit via cloning parent ...\n",
      "\t16.12s\t = Training   runtime\n",
      "\t5.2s\t = Validation runtime\n",
      "Fitting model: WeightedEnsemble_L5_FULL | Skipping fit via cloning parent ...\n",
      "\t0.92s\t = Training   runtime\n",
      "Refit complete, total runtime = 12.9s\n",
      "TabularPredictor saved. To load, use: predictor = TabularPredictor.load(\"AutogluonModels\\ag-20230821_225132\\\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                          model    score_val  pred_time_val    fit_time  pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  fit_order\n",
      "0           WeightedEnsemble_L3  -128.394605      20.428010  161.196516                0.002827           1.068730            3       True          8\n",
      "1          ExtraTreesMSE_BAG_L2  -128.437743      14.648240   60.588385                5.102211          16.640004            2       True          7\n",
      "2           WeightedEnsemble_L4  -130.726208      31.088112  243.651568                0.003007           1.119422            4       True         11\n",
      "3          ExtraTreesMSE_BAG_L3  -130.813888      25.346287  174.420966                4.921104          14.293180            3       True         10\n",
      "4        RandomForestMSE_BAG_L2  -130.892380      15.322972  143.487782                5.776943          99.539400            2       True          6\n",
      "5           WeightedEnsemble_L2  -132.037970       8.184768   44.727038                0.005276           1.758995            2       True          5\n",
      "6        RandomForestMSE_BAG_L3  -132.517257      26.164001  228.238966                5.738818          68.111181            3       True          9\n",
      "7        RandomForestMSE_BAG_L1  -132.567423       4.108985   32.086927                4.108985          32.086927            1       True          3\n",
      "8           WeightedEnsemble_L5  -133.317307      42.391793  330.859310                0.003008           0.917157            5       True         14\n",
      "9          ExtraTreesMSE_BAG_L4  -133.358041      36.282283  258.652357                5.197178          16.120211            4       True         13\n",
      "10       RandomForestMSE_BAG_L4  -135.522252      37.191607  313.821942                6.106502          71.289796            4       True         12\n",
      "11         ExtraTreesMSE_BAG_L1  -167.403368       4.070507   10.881117                4.070507          10.881117            1       True          4\n",
      "12        KNeighborsDist_BAG_L1 -2322.238888       0.684929    0.514048                0.684929           0.514048            1       True          2\n",
      "13        KNeighborsUnif_BAG_L1 -2420.598368       0.681608    0.466289                0.681608           0.466289            1       True          1\n",
      "14   KNeighborsUnif_BAG_L1_FULL          NaN       0.681608    0.466289                0.681608           0.466289            1       True         15\n",
      "15   KNeighborsDist_BAG_L1_FULL          NaN       0.684929    0.514048                0.684929           0.514048            1       True         16\n",
      "16    ExtraTreesMSE_BAG_L1_FULL          NaN       4.070507   10.881117                4.070507          10.881117            1       True         18\n",
      "17  RandomForestMSE_BAG_L1_FULL          NaN       4.108985   32.086927                4.108985          32.086927            1       True         17\n",
      "18    ExtraTreesMSE_BAG_L2_FULL          NaN      14.648240   60.588385                5.102211          16.640004            2       True         21\n",
      "19  RandomForestMSE_BAG_L2_FULL          NaN      15.322972  143.487782                5.776943          99.539400            2       True         20\n",
      "20    ExtraTreesMSE_BAG_L3_FULL          NaN      25.346287  174.420966                4.921104          14.293180            3       True         24\n",
      "21  RandomForestMSE_BAG_L3_FULL          NaN      26.164001  228.238966                5.738818          68.111181            3       True         23\n",
      "22    ExtraTreesMSE_BAG_L4_FULL          NaN      36.282283  258.652357                5.197178          16.120211            4       True         27\n",
      "23  RandomForestMSE_BAG_L4_FULL          NaN      37.191607  313.821942                6.106502          71.289796            4       True         26\n",
      "24     WeightedEnsemble_L5_FULL          NaN            NaN  330.859310                     NaN           0.917157            5       True         28\n",
      "25     WeightedEnsemble_L4_FULL          NaN            NaN  243.651568                     NaN           1.119422            4       True         25\n",
      "26     WeightedEnsemble_L3_FULL          NaN            NaN  161.196516                     NaN           1.068730            3       True         22\n",
      "27     WeightedEnsemble_L2_FULL          NaN            NaN   44.727038                     NaN           1.758995            2       True         19\n",
      "                          model    score_val  pred_time_val    fit_time  \\\n",
      "0           WeightedEnsemble_L3  -128.394605      20.428010  161.196516   \n",
      "1          ExtraTreesMSE_BAG_L2  -128.437743      14.648240   60.588385   \n",
      "2           WeightedEnsemble_L4  -130.726208      31.088112  243.651568   \n",
      "3          ExtraTreesMSE_BAG_L3  -130.813888      25.346287  174.420966   \n",
      "4        RandomForestMSE_BAG_L2  -130.892380      15.322972  143.487782   \n",
      "5           WeightedEnsemble_L2  -132.037970       8.184768   44.727038   \n",
      "6        RandomForestMSE_BAG_L3  -132.517257      26.164001  228.238966   \n",
      "7        RandomForestMSE_BAG_L1  -132.567423       4.108985   32.086927   \n",
      "8           WeightedEnsemble_L5  -133.317307      42.391793  330.859310   \n",
      "9          ExtraTreesMSE_BAG_L4  -133.358041      36.282283  258.652357   \n",
      "10       RandomForestMSE_BAG_L4  -135.522252      37.191607  313.821942   \n",
      "11         ExtraTreesMSE_BAG_L1  -167.403368       4.070507   10.881117   \n",
      "12        KNeighborsDist_BAG_L1 -2322.238888       0.684929    0.514048   \n",
      "13        KNeighborsUnif_BAG_L1 -2420.598368       0.681608    0.466289   \n",
      "14   KNeighborsUnif_BAG_L1_FULL          NaN       0.681608    0.466289   \n",
      "15   KNeighborsDist_BAG_L1_FULL          NaN       0.684929    0.514048   \n",
      "16    ExtraTreesMSE_BAG_L1_FULL          NaN       4.070507   10.881117   \n",
      "17  RandomForestMSE_BAG_L1_FULL          NaN       4.108985   32.086927   \n",
      "18    ExtraTreesMSE_BAG_L2_FULL          NaN      14.648240   60.588385   \n",
      "19  RandomForestMSE_BAG_L2_FULL          NaN      15.322972  143.487782   \n",
      "20    ExtraTreesMSE_BAG_L3_FULL          NaN      25.346287  174.420966   \n",
      "21  RandomForestMSE_BAG_L3_FULL          NaN      26.164001  228.238966   \n",
      "22    ExtraTreesMSE_BAG_L4_FULL          NaN      36.282283  258.652357   \n",
      "23  RandomForestMSE_BAG_L4_FULL          NaN      37.191607  313.821942   \n",
      "24     WeightedEnsemble_L5_FULL          NaN            NaN  330.859310   \n",
      "25     WeightedEnsemble_L4_FULL          NaN            NaN  243.651568   \n",
      "26     WeightedEnsemble_L3_FULL          NaN            NaN  161.196516   \n",
      "27     WeightedEnsemble_L2_FULL          NaN            NaN   44.727038   \n",
      "\n",
      "    pred_time_val_marginal  fit_time_marginal  stack_level  can_infer  \\\n",
      "0                 0.002827           1.068730            3       True   \n",
      "1                 5.102211          16.640004            2       True   \n",
      "2                 0.003007           1.119422            4       True   \n",
      "3                 4.921104          14.293180            3       True   \n",
      "4                 5.776943          99.539400            2       True   \n",
      "5                 0.005276           1.758995            2       True   \n",
      "6                 5.738818          68.111181            3       True   \n",
      "7                 4.108985          32.086927            1       True   \n",
      "8                 0.003008           0.917157            5       True   \n",
      "9                 5.197178          16.120211            4       True   \n",
      "10                6.106502          71.289796            4       True   \n",
      "11                4.070507          10.881117            1       True   \n",
      "12                0.684929           0.514048            1       True   \n",
      "13                0.681608           0.466289            1       True   \n",
      "14                0.681608           0.466289            1       True   \n",
      "15                0.684929           0.514048            1       True   \n",
      "16                4.070507          10.881117            1       True   \n",
      "17                4.108985          32.086927            1       True   \n",
      "18                5.102211          16.640004            2       True   \n",
      "19                5.776943          99.539400            2       True   \n",
      "20                4.921104          14.293180            3       True   \n",
      "21                5.738818          68.111181            3       True   \n",
      "22                5.197178          16.120211            4       True   \n",
      "23                6.106502          71.289796            4       True   \n",
      "24                     NaN           0.917157            5       True   \n",
      "25                     NaN           1.119422            4       True   \n",
      "26                     NaN           1.068730            3       True   \n",
      "27                     NaN           1.758995            2       True   \n",
      "\n",
      "    fit_order  \n",
      "0           8  \n",
      "1           7  \n",
      "2          11  \n",
      "3          10  \n",
      "4           6  \n",
      "5           5  \n",
      "6           9  \n",
      "7           3  \n",
      "8          14  \n",
      "9          13  \n",
      "10         12  \n",
      "11          4  \n",
      "12          2  \n",
      "13          1  \n",
      "14         15  \n",
      "15         16  \n",
      "16         18  \n",
      "17         17  \n",
      "18         21  \n",
      "19         20  \n",
      "20         24  \n",
      "21         23  \n",
      "22         27  \n",
      "23         26  \n",
      "24         28  \n",
      "25         25  \n",
      "26         22  \n",
      "27         19  \n",
      "Best Model: WeightedEnsemble_L3\n",
      "Best Model Info: WeightedEnsemble_L3\n",
      "         num_date_time       answer\n",
      "0        1_20220825 00  2140.416748\n",
      "1        1_20220825 01  2198.230469\n",
      "2        1_20220825 02  1992.831421\n",
      "3        1_20220825 03  2031.187866\n",
      "4        1_20220825 04  2029.455078\n",
      "...                ...          ...\n",
      "16795  100_20220831 19   918.201660\n",
      "16796  100_20220831 20   873.628296\n",
      "16797  100_20220831 21   804.887329\n",
      "16798  100_20220831 22   721.251587\n",
      "16799  100_20220831 23   603.298767\n",
      "\n",
      "[16800 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random\n",
    "import os\n",
    "\n",
    "from tqdm import tqdm\n",
    "from autogluon.tabular import TabularDataset, TabularPredictor\n",
    "\n",
    "\n",
    "train_df = pd.read_csv('./train.csv')\n",
    "test_df = pd.read_csv('./test.csv')\n",
    "sample_submission = pd.read_csv('./sample_submission.csv')\n",
    "building_df = pd.read_csv('./building_info.csv')\n",
    "\n",
    "from datetime import datetime, timedelta\n",
    "\n",
    "def classify_weekdays(start_date, end_date):\n",
    "    date_format = \"%Y-%m-%d\"\n",
    "    start = datetime.strptime(start_date, date_format)\n",
    "    end = datetime.strptime(end_date, date_format)\n",
    "    \n",
    "    classification = []\n",
    "    current_date = start\n",
    "    \n",
    "    while current_date <= end:\n",
    "        if current_date.weekday() < 5:  # 0: 월요일, 1: 화요일, ..., 4: 금요일\n",
    "            classification.append(0)  # 평일\n",
    "        else:\n",
    "            classification.append(1)  # 주말\n",
    "        \n",
    "        current_date += timedelta(days=1)\n",
    "    \n",
    "    return classification\n",
    "\n",
    "# 예시 데이터 프레임 생성\n",
    "data = {'date': pd.date_range(start='2022-06-01', end='2022-08-24')}\n",
    "train = pd.DataFrame(data)\n",
    "t_data = {'date': pd.date_range(start='2022-08-25', end='2022-08-31')}\n",
    "test = pd.DataFrame(t_data)\n",
    "\n",
    "# 주말/평일 분류한 결과를 'holidays' 열에 추가\n",
    "start_date = \"2022-06-01\"\n",
    "end_date = \"2022-08-24\"\n",
    "test_start_date = \"2022-08-25\"\n",
    "test_end_date = \"2022-08-31\"\n",
    "train['holidays'] = classify_weekdays(start_date, end_date)\n",
    "train_df['holidays'] = train['holidays']\n",
    "test['holidays'] = classify_weekdays(test_start_date, test_end_date)\n",
    "test_df['holidays'] = test['holidays']\n",
    "#added\n",
    "train_df['일시'] = pd.to_datetime(train_df['일시'], format='%Y%m%d %H')\n",
    "test_df['일시'] = pd.to_datetime(test_df['일시'], format='%Y%m%d %H')\n",
    "\n",
    "# date time feature 생성\n",
    "train_df['hour'] = train_df['일시'].dt.hour\n",
    "train_df['day'] = train_df['일시'].dt.day\n",
    "train_df['month'] = train_df['일시'].dt.month\n",
    "train_df['year'] = train_df['일시'].dt.year\n",
    "\n",
    "test_df['hour'] = test_df['일시'].dt.hour\n",
    "test_df['day'] = test_df['일시'].dt.day\n",
    "test_df['month'] = test_df['일시'].dt.month\n",
    "test_df['year'] = test_df['일시'].dt.year\n",
    "\n",
    "train_df['sin_time'] = np.sin(2*np.pi*train_df.hour/24)\n",
    "train_df['cos_time'] = np.cos(2*np.pi*train_df.hour/24)\n",
    "\n",
    "test_df['sin_time'] = np.sin(2*np.pi*test_df.hour/24)\n",
    "test_df['cos_time'] = np.cos(2*np.pi*test_df.hour/24)\n",
    "train_df['THI'] = 9/5*train_df['기온(C)'] - 0.55*(1-train_df['습도(%)']/100)*(9/5*train_df['습도(%)']-26)+32\n",
    "test_df['THI'] = 9/5*test_df['기온(C)'] - 0.55*(1-test_df['습도(%)']/100)*(9/5*test_df['습도(%)']-26)+32\n",
    "##############################################################\n",
    "\n",
    "\n",
    "train_df = train_df.drop(columns = ['일조(hr)', '일사(MJ/m2)', '강수량(mm)', '풍속(m/s)'])\n",
    "test_df = test_df.drop(columns = ['강수량(mm)', '풍속(m/s)'])\n",
    "train_data = TabularDataset(train_df)\n",
    "train_data.head()\n",
    "\n",
    "label = '전력소비량(kWh)'\n",
    "####################################################\n",
    "\n",
    "# AutoGluon 모델 학습 및 결과 저장\n",
    "predictor = TabularPredictor(label=label, problem_type='regression').fit(\n",
    "                            train_data, presets=['best_quality'], auto_stack=True,\n",
    "                            num_bag_folds=10, num_bag_sets=20, num_stack_levels=3, refit_full=True)\n",
    "\n",
    "# 가장 좋은 모델 정보 출력\n",
    "leaderboard = predictor.leaderboard()\n",
    "print(leaderboard)\n",
    "\n",
    "# 선택된 최적 모델 정보 출력\n",
    "best_model = leaderboard.iloc[0]['model']\n",
    "print(\"Best Model:\", best_model)\n",
    "\n",
    "# 선택된 최적 모델의 성능 정보 출력\n",
    "best_model_info = predictor.get_model_best()\n",
    "print(\"Best Model Info:\", best_model_info)\n",
    "\n",
    "test_data = TabularDataset(test_df)\n",
    "\n",
    "y_pred = predictor.predict(test_data)\n",
    "x = pd.DataFrame(y_pred)\n",
    "sample_submission['answer'] = x['전력소비량(kWh)']\n",
    "print(sample_submission)\n",
    "sample_submission.to_csv('./baseline_submission_org.csv', index=False)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
